ted2zim/scraper.py

```diff
@@ -168,31 +198,38 @@ def extract_all_video_links(self):
         self.update_title_and_description()
 
     def update_title_and_description(self):
-        if len(self.topics) > 1:
+        if self.playlist:
             if not self.title:
-                self.title = "TED Collection"
+                self.title = "TED Playlist"
             if not self.description:
-                self.description = "A selection of TED videos from several topics"
+                self.description = "A curated list of related TED videos"
         else:
-            if not self.title:
-                topic_str = self.topics[0].replace("+", " ")
-                self.title = f"{topic_str.capitalize()} from TED"
-            if not self.description:
-                self.description = f"A selection of {topic_str} videos from TED"
+            if len(self.topics) > 1:
+                if not self.title:
+                    self.title = "TED Collection"
+                if not self.description:
+                    self.description = "A selection of TED videos from several topics"
+            else:
+                if not self.title:
+                    topic_str = self.topics[0].replace("+", " ")
+                    self.title = f"{topic_str.capitalize()} from TED"
+                if not self.description:
+                    self.description = f"A selection of {topic_str} videos from TED"
 
-    def extract_videos(self, video_allowance):
+    def extract_videos_on_current_page(self, page_html, video_allowance):
```

Comment:
remove the `current` part of that method name as you're passing the page's HTML to extract from so there's no such thing as `current`.

Reply:
Done